lxd/db/images.go

```diff
@@ -1138,6 +1138,17 @@ WHERE images.fingerprint = ?
 	return c.getNodesByImageFingerprint(q, fingerprint)
 }
 
+// GetNodesWithImageAndAutoUpdate returns the addresses of online nodes which already have the image.
+func (c *Cluster) GetNodesWithImageAndAutoUpdate(fingerprint string, autoUpdate bool) ([]string, error) {
+	q := `
+SELECT DISTINCT nodes.address FROM nodes
+  LEFT JOIN images_nodes ON images_nodes.node_id = nodes.id
+  LEFT JOIN images ON images_nodes.image_id = images.id
+WHERE images.fingerprint = ? AND images.auto_update = ?
+	`
+	return c.getNodesByImageFingerprintAndAutoUpdate(q, fingerprint, autoUpdate)
```

Comment:
@monstermunchkin @stgraber I can see this pattern of defining the query in the exported function and then passing it to the unexported function to actually run the query is already in use in this file, but is there a reason that the contents of `getNodesByImageFingerprintAndAutoUpdate` couldn't just be in this function as its not used elsewhere?

Reply:
@tomponline I removed `getNodesByImageFingerprintAndAutoUpdate` and use `getNodesByImageFingerprint` instead which now takes a `*bool` as well.